pkg/resource/stack/deployment.go

```diff
@@ -428,11 +443,8 @@ func DeserializeDeploymentV3(
 		ops = append(ops, desop)
 	}
 
-	if completeBatch != nil {
```

Comment:
The new `initializeSecretsManagerAndDecrypter` now always returns a non-nil `completeBatch` function, so no need to check for nil anymore.